### DISCUSSION TITLE
Stashed Windows Persist When Using Stage Manager

### DISCUSSION THREAD

**Discussion by vcbrewers** [rxhanson/RectanglePro-Community#542] [Q&A]
### Issue

- When utilizing Stage Manager, Stashed Windows persist when switching to an app group ("stage") that does _not_ include the stashed app(s).
- Scrolling over a Stashed Window from an inactive stage causes the entire stage to switch.

### Example

- App Group/Stage 1: Reeder (left 2/3rds), Goodlinks (stashed - last 1/3) 
- App Group/Stage 2: Apple News (left 2/3rds)

- When switching from Stage 1 to Stage 2, Goodlinks remains stashed on screen and scrolling over it switches the app grouping from Group 2 to Group 1.

### Question/Request

- [ ] Is it possible to associate a stashed window(s) with a specific stage instead of having it be persistent across stages? OR
- [ ] Is it possible to have stashed window(s) remain persistent but be associated with _all_ stages within a Space, so that these apps can be utilized without causing the app group/stage to change? 

Apologies in advance if this is very convoluted. More than happy to try to make a screen recording if that would help.

macOS version - Sequoia 15.1 Developer Beta

**Comment by rxhanson** [comment]
Thanks for the detailed feedback.

> Is it possible to associate a stashed window(s) with a specific stage instead of having it be persistent across stages? OR

> Is it possible to have stashed window(s) remain persistent but be associated with all stages within a Space, so that these apps can be utilized without causing the app group/stage to change?

There isn't an API for interacting with Stage Manager, so there's not a great way to achieve either of these, unfortunately. Technically, it could be possible to do the first option, although without an API it would be prone to breaking with each macOS release. I'm not sold on adding something in for it at this time, but would consider taking another look with significant demand.
